velox/vector/FlatVector-inl.h

```diff
@@ -391,5 +382,80 @@ void FlatVector<T>::prepareForReuse() {
     rawValues_ = nullptr;
   }
 }
+
+template <typename T>
+void FlatVector<T>::resizeValues(
+    BufferPtr* buffer,
+    velox::memory::MemoryPool* pool,
+    vector_size_t newSize,
+    const std::optional<T>& initialValue) {
+  if (buffer->get() && buffer->get()->unique() && buffer->get()->isMutable()) {
+    const uint64_t newByteSize = BaseVector::byteSize<T>(newSize);
+    if (buffer->get()->size() < newByteSize) {
+      AlignedBuffer::reallocate<T>(buffer, newSize, initialValue);
+    } else {
+      buffer->get()->setSize(newByteSize);
+    }
+    return;
+  }
+  BufferPtr newValues;
+  newValues = AlignedBuffer::allocate<T>(newSize, pool, initialValue);
+
+  if (buffer->get()) {
+    if constexpr (Buffer::is_pod_like_v<T>) {
+      auto dst = newValues->asMutable<T>();
+      auto src = buffer->get()->as<T>();
+      auto len = std::min(buffer->get()->size(), newValues->size());
+      memcpy(dst, src, len);
+    } else {
+      auto previousSize = BaseVector::length_;
+      auto rawOldValues = newValues->asMutable<T>();
+      auto rawNewValues = newValues->asMutable<T>();
+      SelectivityVector rowsToCopy(
+          std::min<vector_size_t>(newSize, previousSize));
```

Comment:
nit: creating SelectivityVectors is not free. It would be more efficient to use a for loop from 0 to std::min...